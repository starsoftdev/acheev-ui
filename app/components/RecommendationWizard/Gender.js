// @flow
import * as React from 'react';

import BaseModal from 'components/BaseModal';
import Button from 'components/Button';
import Icon from 'components/Icon';
import MaleIcon from 'images/sprite/male.svg';
import FemaleIcon from 'images/sprite/female.svg';
import OtherIcon from 'images/sprite/other_gender.svg';

import Block from './Block';

const options = [
  {
    icon: MaleIcon,
    label: 'Male',
    field: 'male',
  },
  {
    icon: FemaleIcon,
    label: 'Famale',
    field: 'female',
  },
  {
    icon: OtherIcon,
    label: 'Other',
    field: 'other',
  },
];

type Props = {
  onSelect: (value: string) => void,
  // eslint-disable-next-line  react/no-unused-prop-types
  value: string | Array<string>,
};

type State = {
  value?: string,
  isModalOpen: boolean,
};

export default class Gender extends React.Component<Props, State> {
  state = {
    value: undefined,
    isModalOpen: false,
  };

  onOpenModal = (e: SyntheticEvent<HTMLAnchorElement>) => {
    e.preventDefault();
    this.setState({ isModalOpen: true });
  };

  onCloseModal = () => {
    this.setState({ isModalOpen: false });
  };

  handleClick = (e: SyntheticEvent<HTMLInputElement>) => {
    const { name } = e.currentTarget.dataset;
    this.setState({ value: name });
    this.props.onSelect(name);
  };

  render() {
    return (
      <section className="recommendationWizard__body">
        <div className="container">
          <div className="row">
            <h1 className="recommendationWizard__title small-12 column">
              What is your Gender?
            </h1>
            <div className="small-12 column">
              <div className="row">
                <div className="small-12 large-10 large-offset-1 column">
                  <div className="row">
                    {options.map(item => (
                      <div key={item.field} className="small-12 large-4 column">
                        <Block
                          label={item.label}
                          onClick={this.handleClick}
                          data-name={item.field}
                          selected={item.field === this.state.value}
                        >
                          <Icon
                            glyph={item.icon}
                            width={100}
                            height={35}
                            className="recommendationWizard__icon"
                          />
                        </Block>
                      </div>
                    ))}
                  </div>
                </div>

                <div className="small-12 column">
                  <p className="recommendationWizard--center">
                    <a
                      href="/recommender/dumb-link"
                      className="recommendationWizard__link"
                      onClick={this.onOpenModal}
                    >
                      Why do we ask for gender?
                    </a>
                  </p>
                </div>
              </div>
            </div>
          </div>
        </div>

        <BaseModal
          className="small-12 medium-6 column"
          isOpen={this.state.isModalOpen}
        >
          <h2>Why do we ask for gender?</h2>
          <p className="recommendationWizard__modalBody">
            Lift requests your gender to help to customize your cannabis product
            suggestion based on similar users within our database.
          </p>

          <div className="small-12 small-centered text-center columns">
            <Button onClick={this.onCloseModal}>Got it!</Button>
          </div>
        </BaseModal>
      </section>
    );
  }
}
