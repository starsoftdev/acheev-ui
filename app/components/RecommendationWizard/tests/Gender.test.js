import React from 'react';
import { create } from 'react-test-renderer';
import { shallow } from 'enzyme';
import Gender from 'components/RecommendationWizard/Gender';

describe('<Gender />', () => {
  it('should match snapshot', () => {
    const component = create(<Gender onSelect={() => {}} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Should call onSelect', () => {
    const handleSelect = jest.fn();
    const component = shallow(<Gender onSelect={handleSelect} />);

    component
      .find('Block')
      .first()
      .simulate('click', { currentTarget: { dataset: { name: 'new' } } });

    expect(handleSelect).toHaveBeenCalledWith('new');
    expect(handleSelect).toHaveBeenCalledTimes(1);
  });
});
