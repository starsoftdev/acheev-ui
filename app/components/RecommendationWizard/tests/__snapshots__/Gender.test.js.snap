// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Gender /> should match snapshot 1`] = `
<section
  className="recommendationWizard__body"
>
  <div
    className="container"
  >
    <div
      className="row"
    >
      <h1
        className="recommendationWizard__title small-12 column"
      >
        What is your Gender?
      </h1>
      <div
        className="small-12 column"
      >
        <div
          className="row"
        >
          <div
            className="small-12 large-10 large-offset-1 column"
          >
            <div
              className="row"
            >
              <div
                className="small-12 large-4 column"
              >
                <div
                  className="recommendationWizard__block"
                  data-name="male"
                  onClick={[Function]}
                >
                  <svg
                    className="recommendationWizard__icon"
                    height={35}
                    width={100}
                  >
                    <use
                      xlinkHref="#undefined"
                    />
                  </svg>
                  <p
                    className="recommendationWizard__label"
                  >
                    Male
                  </p>
                </div>
              </div>
              <div
                className="small-12 large-4 column"
              >
                <div
                  className="recommendationWizard__block"
                  data-name="female"
                  onClick={[Function]}
                >
                  <svg
                    className="recommendationWizard__icon"
                    height={35}
                    width={100}
                  >
                    <use
                      xlinkHref="#undefined"
                    />
                  </svg>
                  <p
                    className="recommendationWizard__label"
                  >
                    Famale
                  </p>
                </div>
              </div>
              <div
                className="small-12 large-4 column"
              >
                <div
                  className="recommendationWizard__block"
                  data-name="other"
                  onClick={[Function]}
                >
                  <svg
                    className="recommendationWizard__icon"
                    height={35}
                    width={100}
                  >
                    <use
                      xlinkHref="#undefined"
                    />
                  </svg>
                  <p
                    className="recommendationWizard__label"
                  >
                    Other
                  </p>
                </div>
              </div>
            </div>
          </div>
          <div
            className="small-12 column"
          >
            <p
              className="recommendationWizard--center"
            >
              <a
                className="recommendationWizard__link"
                href="/recommender/dumb-link"
                onClick={[Function]}
              >
                Why do we ask for gender?
              </a>
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</section>
`;
